src/components/filters/filter-panels/new-filter-panel.js

```diff
@@ -21,12 +21,10 @@
 import React, {useCallback} from 'react';
 import {StyledFilterContent} from 'components/common/styled-components';
 import FilterPanelHeaderFactory from 'components/side-panel/filter-panel/filter-panel-header';
-import FieldSelector from 'components/common/field-selector';
 import SourceDataSelectorFactory from 'components/side-panel/common/source-data-selector';
+import FieldSelectorFactory from '../../common/field-selector';
 
-NewFilterPanelFactory.deps = [FilterPanelHeaderFactory, SourceDataSelectorFactory];
-
-function NewFilterPanelFactory(FilterPanelHeader, SourceDataSelector) {
+const NewFilterPanelFactory = (FilterPanelHeader, SourceDataSelector, FieldSelector) => {
```

Comment:
make `NewFilterPanelFactory` a actual function, instead of `const ... = () => {}` so you won't have errors like `xxx is not declared before it is being used`

make it

```
function NewFilterPanelFactory(...) {
...
}
```

Reply:
Done